pymodm/base/options.py

```diff
@@ -75,23 +77,41 @@ def auto_dereference(self, auto_dereference):
         self._auto_dereference = auto_dereference
 
     def get_field(self, field_name):
-        """Retrieve a Field instance with the given name."""
+        """Retrieve a Field instance with the given MongoDB name."""
         return self.fields_dict.get(field_name)
 
+    def get_field_from_attname(self, attname):
+        """Retrieve a Fields instance with the given attribute name."""
+        return self.fields_attname_dict.get(attname)
+
+    def _add_field(self, field_inst):
+        """Helper for add_field."""
+        self.fields_dict[field_inst.mongo_name] = field_inst
+        self.fields_attname_dict[field_inst.attname] = field_inst
+        index = bisect(self.fields_ordered, field_inst)
+        self.fields_ordered.insert(index, field_inst)
+
     def add_field(self, field_inst):
         """Add or replace a Field with a given name."""
```

Comment:
Nit (my fault): there is no name given in this method. Only the field instance is provided. This should read `Add or replace a given Field.` Since this docstring doesn't appear in the public documentation, there's no need for markup.